homeassistant/components/climate/daikin.py

```diff
@@ -0,0 +1,275 @@
+"""
+Support for the Daikin HVAC.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/climate.daikin/
+"""
+import logging
+import voluptuous as vol
+
+from homeassistant.components.climate import (
+    ClimateDevice,
+    ATTR_OPERATION_MODE,
+    SUPPORT_TARGET_TEMPERATURE, SUPPORT_TARGET_HUMIDITY,
+    SUPPORT_FAN_MODE,
+    SUPPORT_OPERATION_MODE, SUPPORT_SWING_MODE,
+    PLATFORM_SCHEMA)
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME,
+    TEMP_CELSIUS, ATTR_TEMPERATURE)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pydaikin==0.4']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=None): cv.string,
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE | SUPPORT_TARGET_HUMIDITY |
+                 SUPPORT_FAN_MODE |
+                 SUPPORT_OPERATION_MODE |
+                 SUPPORT_SWING_MODE)
+
+DAIKIN_STATE_TO_HA = {
+    'fan': 'Fan only',
+    'dry': 'Dry',
+    'cool': 'Cool',
+    'hot': 'Heat',
+    'off': 'Off',
+    'auto': 'Auto',
+    'auto-1': 'Auto',
+    'auto-7': 'Auto'
+}
+
+HA_STATE_TO_DAIKIN = {value: key for key, value in DAIKIN_STATE_TO_HA.items()}
```

Comment:
This is not well-defined: Since you have `Auto` as a value multiple times, `HA_STATE_TO_DAIKIN['Auto']` may be any of `auto`, `auto-1` or `auto-7`, depending on the order of the iteration. Since dicts are not ordered, this is not deterministic.